velox/functions/prestosql/DecimalArithmetic.cpp

```diff
@@ -97,6 +97,60 @@ class DecimalBaseFunction : public exec::VectorFunction {
   const uint8_t bRescale_;
 };
 
+template <
+    typename R /* Result Type */,
+    typename A /* Argument */,
+    typename Operation /* Arithmetic operation */>
+class DecimalUnaryBaseFunction : public exec::VectorFunction {
+ public:
+  DecimalUnaryBaseFunction(uint8_t aRescale) : aRescale_(aRescale) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& resultType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    auto rawResults = prepareResults(rows, resultType, context, result);
+    if (args[0]->isConstantEncoding()) {
+      // Fast path for (const, flat).
```

Comment:
Shouldn't this say "Fast path for constant", since there is only one argument?

Same below.